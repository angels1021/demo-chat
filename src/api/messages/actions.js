/**
 * Action creators for the auth api.
 * @module path/to/file/actions
 * @typicalname actions
 * @requires [redux-actions](https://www.npmjs.com/package/redux-actions)
 * */

import { ADD_MESSAGE } from './constants';

/**
 * An action object.
 * @typedef {{type:(string),payload:(object)}} action
 * */

/**
 * @function addMessage
 * @memberOf module:path/to/file/actions
 * @static
 *
 * @returns {action} action object - { type: ADD_MESSAGE, payload: {} }
 */
export const addMessage = (message) => ({
  type: ADD_MESSAGE,
  payload: message
});
