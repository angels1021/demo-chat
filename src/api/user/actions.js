import uniqueId from 'uniqid';
import getProfileImage from '../../utils/profileImages';
import { CREATE_USER } from './constants';

export const createUser = (username) => {
  const id = uniqueId('user_');
  return {
    type: CREATE_USER,
    payload: { id, username, avatar: getProfileImage() }
  };
};
