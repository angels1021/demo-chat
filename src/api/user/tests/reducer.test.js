/**
 * test reducer
 **/
import userReducer from '../reducer';
import { createUser } from '../actions';

describe('api/user/reducer', () => {
  let state;
  beforeEach(() => {
    state = userReducer();
  });

  it('should return the initial state', () => {
    expect(state).toMatchSnapshot();
  });

  describe('CREATE_USER', () => {
    it('should set a new user', () => {
      // arrange
      const action = createUser('tom');
      // act
      const result = userReducer(state, action);
      // assert
      expect(result).toEqual(action.payload);
    });
  });
});
