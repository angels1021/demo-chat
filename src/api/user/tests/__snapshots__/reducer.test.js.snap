// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/user/reducer should return the initial state 1`] = `
Object {
  "avatar": "",
  "id": "",
  "username": "",
}
`;
